utils/helpers/userHelper.ts

```diff
@@ -0,0 +1,53 @@
+import {UserProfile} from "@auth0/nextjs-auth0/src/frontend/use-user";
+import {db} from "../../firebase";
+
+export const getUserFromFirebase = async (user: UserProfile) => {
+    try {
+        //creates user if not created.
+        db.collection("auth0").where('auth0', '==', user.sub).get().then((data) => {
+            if (!data.docs.length) {
+                // to create Auth0 of the new user.
+                db.collection('auth0').doc().set({
```

Comment:
In my opinion we should start abstracting the different types of database calls into helper functions and importing them where needed.

In this function we have set calls to three different collections with roughly the same syntax. It would pay dividends later to have three wrapper functions:

`setUserDoc`, `setAuth0Doc`, and `setProfileDoc` where all we pass in is the id and data struct.

Federico mentions the firebase v9 syntax which goes a long way towards modularizing the code. Maybe that's sufficient. 

But we make these types calls so much that nicely named wrappers will help make the code base more intuitive and alleviate the need for some comments.